lint/range-expr-copy_checker.go

```diff
@@ -0,0 +1,52 @@
+package lint
+
+import (
+	"go/ast"
+	"go/types"
+)
+
+// rangeExprCopyCheck detects for statements with range expressions
+// that perform excessive copying (big arrays can cause it).
+//
+// Suggested fix is to use & and doing a range over pointer to array.
+//
+// Rationale: performance.
+func rangeExprCopyCheck(ctx *context) func(*ast.File) {
+	// TODO(quasilyte): there is some annoying code duplication with other
+	// range statement checker. We should consider refactoring if
+	// more checkers that inspect range statements will appear.
+	return wrapStmtChecker(&rangeExprCopyChecker{
+		baseStmtChecker: baseStmtChecker{ctx: ctx},
+	})
+}
+
+type rangeExprCopyChecker struct {
+	baseStmtChecker
+}
+
+func (c *rangeExprCopyChecker) PerFuncInit(fn *ast.FuncDecl) bool {
+	return fn.Body != nil && !c.ctx.IsUnitTestFuncDecl(fn)
```

Comment:
```c.ctx.IsUnitTestFuncDecl(fn)``` why is it necessary?

Reply:
Because many tests use table-driven approach and it leads to kind of false-positive warnings.